packages/components/autocomplete/src/autocomplete.vue

```diff
@@ -246,7 +246,11 @@ const handleFocus = (evt: FocusEvent) => {
     emit('focus', evt)
 
     if (props.triggerOnFocus && !readonly) {
-      debouncedGetData(String(props.modelValue))
+      const stringfiedValue =
+        props.modelValue === null || props.modelValue === undefined
+          ? props.modelValue
+          : String(props.modelValue)
+      debouncedGetData(stringfiedValue)
```

Comment:
@fayeah I think the modification according to @nova1751 is more concise.
https://github.com/element-plus/element-plus/pull/15728/files#diff-b8f36a65dcdc4f9c18cbc8b2fa5201f7c02df8c39b90552ffe7ee61b191a2948R248-R249